File: Visualizer/ShaCalc/Model/ConstantBit.cs

namespace ShaCalc.Model
{
    class ConstantBit : BitValue
    {
        public bool Variable;

        public ConstantBit(bool value)
        {
            _value = value;
            _isCalc = true;
        }

        protected override bool? Calc()
        {
            return _value;
        }

        protected override bool Request(bool value, bool hard)
        {
            return _value.Value == value;
        }

        public override BitValue[] GetInputs()
        {
            return null;
        }

        public override string GetName()
        {
            return _value.HasValue ? (_value.Value ? "1" : "0") : "?";
        }
        
        protected override BitValue DoOptimize()
        {
            return this;
        }
    }
}

File: Graphula/Graphula/constantbit.cpp
#include "constantbit.h"

ConstantBit::ConstantBit(bool value)
{
    _isCalc = true;
    _value = value;
}


BitValue *ConstantBit::GetInputs()
{
    return nullptr;
}

int ConstantBit::GetInputsCount()
{
    return 0;
}

QString ConstantBit::GetName()
{
    if (_value) return "1";
    return "0";
}

bool ConstantBit::Calc()
{
    return _value;
}
